File: frontend/src/constants/reviewReasons.ts
/** Синхронизировано с backend/app/schemas/review_reasons.py */
export const REVIEW_REASON_OPTIONS: { id: string; label: string }[] = [
  { id: "insufficient_context", label: "Недостаточно контекста" },
  { id: "wrong_risk_level", label: "Неверный уровень риска" },
  { id: "weak_citations", label: "Слабые цитаты / источники" },
  { id: "bad_draft_tone", label: "Тон / стиль черновика" },
  { id: "incorrect_legal_basis", label: "Неверная правовая база" },
  { id: "too_generic", label: "Слишком общий ответ" },
  { id: "hallucination_suspected", label: "Подозрение на галлюцинацию" },
  { id: "formatting_problem", label: "Форматирование" },
  { id: "operator_preferred_manual", label: "Предпочтительно вручную" },
  { id: "bad_price_range", label: "Цена / диапазон не рынок" },
  { id: "other", label: "Другое" },
];

File: backend/app/schemas/review_reasons.py
from __future__ import annotations

REVIEW_REASON_CODES: frozenset[str] = frozenset(
    {
        "insufficient_context",
        "wrong_risk_level",
        "weak_citations",
        "bad_draft_tone",
        "incorrect_legal_basis",
        "too_generic",
        "hallucination_suspected",
        "formatting_problem",
        "operator_preferred_manual",
        "bad_price_range",
        "other",
    }
)


def normalize_reason_codes(raw: list[str] | None) -> list[str]:
    """Фильтрация неизвестных кодов, порядок сохраняется, дубликаты убираются."""
    if not raw:
        return []
    seen: set[str] = set()
    out: list[str] = []
    for x in raw:
        s = str(x).strip()
        if not s or s not in REVIEW_REASON_CODES:
            continue
        if s not in seen:
            seen.add(s)
            out.append(s)
    return out
